Source/WebCore/platform/graphics/cocoa/MultiRepresentationHEICRenderingUtilities.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(MULTI_REPRESENTATION_HEIC)
+
+namespace WebCore {
+
+enum class ImageDrawResult;
+
+class FloatSize;
+class Font;
+class GraphicsContext;
+class Image;
+class LayoutUnit;
+
+FloatSize sizeForMultiRepresentationHEIC(const Font&);
+
+float baselineOffsetForMultiRepresentationHEIC(const Font&);
```

Comment:
1. I think `baselineOffset` is a little confusing. Does it mean the offset from top of the line or the bottom of the line? In the fonts' world they use `ascent` and `descent`. Can we use one of them based on what you mean by `baselineOffset`?
2. What is the difference between the Font `ascent/descent` and this `baselineOffset`?
3. Can't we name it `descentForCustomGlyph`?